pkg/util/httpstream/spdy/roundtripper_test.go

```diff
@@ -203,6 +225,21 @@ func TestRoundTripAndNewConnection(t *testing.T) {
 	}
 }
 
+type MockProxyDialer struct {
+	dialer Dialer
+}
+
+func (m *MockProxyDialer) dial(url *url.URL) (net.Conn, error) {
+	if url.Scheme == "https" && url.Host == "proxy.com" {
+		// TODO mock a proxy dial
```

Comment:
@liggitt in case we keep the strategy, any advice here? Just dial to the actual server url?